app/uploaders/alert_image_uploader.rb

```diff
@@ -0,0 +1,57 @@
+class AlertImageUploader < ApplicationUploader
+  include CarrierWave::MiniMagick
+  include ::CarrierWave::Backgrounder::Delay
+
+  def store_dir
+    "#{base_store_dir}/#{model.id}"
+  end
+
+  def base_store_dir
+    "uploads/#{model.class.to_s[0, 2]}"
+  end
+
+  def filename
+    "alert_#{model.id}.jpg"
+  end
+
+  def extension_white_list
+    %w(jpg jpeg gif png tiff tif)
+  end
+
+  process :generate_alert_image
+  process convert: "jpg"
+
+  delegate :bike, to: :model
+
+  def bike_image
+    public_image = bike.public_images.first&.image
+    raise ArgumentError, "No bike image for Bike #{bike.id}" if public_image.blank?
+    public_image
+  end
+
+  def bike_url
+    "bikeindex.org/bikes/#{bike.id}"
+  end
+
+  def bike_location
+    city = bike.registration_address[:city]&.titleize
+    state = bike.registration_address[:state]&.upcase
+
+    if city && state
+      "#{city}, #{state}"
+    elsif state
+      state
+    else
+      ""
+    end
+  end
+
+  def generate_alert_image
+    AlertImageGenerator.generate_image(
+      bike_image_path: bike_image.path,
```

Comment:
I _believe_ this is going to be problematic with images on s3. 

Ideally this with work correctly when locally (like this), but also work correctly for images that are stored on s3 (which still return a relative path, but the path `"uploads/Pu/229/IMG_3624.JPG"` doesn't correspond to a local file.

Reply:
Hmm thanks — I should be able to test this locally with the prod carrierwave settings 👍

Reply:
Worked around this by setting asset host in environment config (development.rb / test.rb / production.rb) and using that value for carrierwave:

```rb
# config/environments/development.rb L57-58 (c3eee6a6)

# Enable serving of images, stylesheets, and JavaScripts from an asset server.
config.action_controller.asset_host = ENV["BASE_URL"]
```
<sup>[[source](https://github.com/bikeindex/bike_index/blob/c3eee6a6/config/environments/development.rb#L57-L58)]</sup>


```rb
# config/environments/production.rb L67-68 (c3eee6a6)

# Enable serving of images, stylesheets, and JavaScripts from an asset server.
config.action_controller.asset_host = ENV["S3_ASSET_HOST"]
```
<sup>[[source](https://github.com/bikeindex/bike_index/blob/c3eee6a6/config/environments/production.rb#L67-L68)]</sup>


```rb
# config/initializers/carrierwave.rb L36 (1a242e11)

config.asset_host = ActionController::Base.asset_host
```
<sup>[[source](https://github.com/bikeindex/bike_index/blob/1a242e11/config/initializers/carrierwave.rb#L36-L36)]</sup>

```rb
bike.alert_image_url
# "http://localhost:3001/uploads/St/72666/flying_pigeon-alert.jpg"
```